extras/parse_travis_log.py

```diff
@@ -20,396 +20,971 @@
+    """Read the NEST Travis CI build log file and return the number of build
+    error and warning messages as well as dictionaries summarizing their
+    occurrences.
+
+    Parameters
+    ----------
+    log_filename:           NEST Travis CI build log file name.
+    msg_make_section_start: Message number string, e.g. "MSGBLD1234".
+    msg_make_section_end:   Message number string, e.g. "MSGBLD1234".
+
+    Returns
+    -------
+    True or False depending on the number of error messages.
+    Number of error messages.
+    Dictionary of file names and the number of errors within these files.
+    Number of warning messages.
+    Dictionary of file names and the number of warnings within these file.
     """
-    assert(clang_format.keys() == cppcheck.keys())
-    assert(clang_format.keys() == vera.keys())
-
-    for filename in clang_format.keys():
-        print(INDENT + ' --' + '-' * len(filename))
-        print(INDENT + ' | ' + filename + ':')
-        print(INDENT + ' | ' + INDENT + ' *  vera++:')
-        for k, v in vera[filename].iteritems():
-            print(INDENT + ' | ' + 2 * INDENT +
-                  ' - ' + k + ' : ' + str(v))
-        print(INDENT + ' | ' + INDENT + ' * cppcheck:')
-        for k, v in cppcheck[filename].iteritems():
-            print(INDENT + ' | ' + 2 * INDENT +
-                  ' - ' + k + ' : ' + str(v))
-        print(INDENT + ' | ' + INDENT + ' * clang_format:')
-        for k, v in clang_format[filename].iteritems():
-            print(INDENT + ' | ' + 2 * INDENT +
-                  ' - ' + k + ' : ' + str(v))
-
-
-def count_warnings_errors(f):
+
+    error_summary = None
+    warning_summary = None
+    number_of_error_msgs = 0
+    number_of_warning_msgs = 0
+    in_make_section = False
+    with open(log_filename) as fh:
+        for line in fh:
+            if is_message(line, msg_make_section_start):
+                in_make_section = True
+                error_summary = {}
+                warning_summary = {}
+
+            if in_make_section:
+                if ': error:' in line:
+                    file_name = line.split(':')[0]
+                    if file_name not in error_summary:
+                        error_summary[file_name] = 0
+                    error_summary[file_name] += 1
+                    number_of_error_msgs += 1
+
+                if ': warning:' in line:
+                    file_name = line.split(':')[0]
+                    if file_name not in warning_summary:
+                        warning_summary[file_name] = 0
+                    warning_summary[file_name] += 1
+                    number_of_warning_msgs += 1
+
+            if is_message(line, msg_make_section_end):
+                # The log file contains only one 'make' section, return.
+                if number_of_error_msgs == 0:
+                    return True, number_of_error_msgs, error_summary, \
+                           number_of_warning_msgs, warning_summary
```

Comment:
For a uniform syntax I added brackest to all other return-statements too. Commit: 36136a4